### PR TITLE
Fix PHP 8.4 deprecation.

### PR DESCRIPTION
**Pull request by kagg-design**
Fixed: "Deprecated: str_getcsv(): the $escape parameter must be provided as its default value will change."

### CODE REVIEW

**Review by jrfnl — COMMENTED** [review]
Nice find!

My two pennies:

I think the `@noinspection` comment should probably be removed as that is an IDE specific thing.

I'd also like to see at least one test covering this change to safeguard it for the future.

**Comment by kagg-design** [comment]
Hi @jrfnl! Thank you for your comment.

I have removed `// @noinspection ...` line.

Concerning tests, I have the following thoughts.

1. This is pure refactoring.
2. As far as I can see, this method was not covered by tests, so should I write a new test for it?
3. This line is called 19 times during tests of the Function Mocker; you can see [my PR](https://github.com/lucatume/function-mocker/pull/21) with 8.4 updates for it. Should it be considered as enough check?

Thank you.

**Comment by jrfnl** [comment]
> Concerning tests, I have the following thoughts.
> 1. This is pure refactoring.
> 2. As far as I can see, this method was not covered by tests, so should I write a new test for it?
> 3. This line is called 19 times during tests of the Function Mocker; you can see [my merged PR](https://github.com/lucatume/function-mocker/pull/21) with 8.4 updates for it. Should it be considered as enough check?

@kagg-design Sorry, but no, tests in an external project should - IMO - definitely not be considered enough of a check.

The existing tests for this package are currently passing on PHP 8.4, so this means this code is currently **not** covered by tests. And even though I checked both the tests and did various scans on the code base with tooling when preparing for PHP 8.4 (see #157, #160, #161, #165, #166), this one wasn't found as it was "hidden" in a callback string.

Considering PHP intends to make further changes to this function in future PHP versions (based on the [RFC](https://wiki.php.net/rfc/deprecations_php_8_4#deprecate_proprietary_csv_escaping_mechanism) and discussions I've had about this in the PHP src GH PR and the Internals mailing list), I think it is only prudent to make sure the code is covered by tests in _this_ project to be sure that, when the next change lands in PHP, we can verify things continue to work as expected.

Aside from that, having a test in place will also safeguard that someone doesn't accidentally removed the fix at some point in the future.

File: tests/get-cached-path.phpt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+echo $actualPath === $expectedPath ? 'PASS' : 'FAIL';
+echo "\n";
+
+// Second pass - index.csv file exists.
+$actualPath = \Patchwork\CodeManipulation\getCachedPath($file);
+
+echo $actualPath === $expectedPath ? 'PASS' : 'FAIL';
+?>
+
+--EXPECT--
```
Comment (jrfnl) [inline tests/get-cached-path.phpt:37]: Should this test get a `--CLEAN--` section to automatically delete the file created during the test after it has run ?

Comment (antecedent) [reply tests/get-cached-path.phpt:37]: Good catch. I think it should.

```suggestion
--CLEAN--
<?php

unlink(__DIR__ . '/cache/index.csv');
rmdir(__DIR__ . '/cache');

?>
--EXPECT--
```

File: tests/get-cached-path.phpt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+use Patchwork\CodeManipulation\Stream;
+
+$cachePath = str_replace('\\', '/', __DIR__ . '/cache');
+Patchwork\CodeManipulation\State::$cacheIndexFile = null;
+Patchwork\Config\State::$cachePath = $cachePath;
+
+if ( ! is_dir($cachePath)) {
+    mkdir($cachePath);
+}
```
Comment (antecedent) [inline tests/get-cached-path.phpt:17]: Personally, I would like to enforce an assumption that the tests leave the file system unaltered.

```suggestion
} else {
    echo "The cache directory should not exist in advance.\n";
}
```

File: tests/get-cached-path.phpt [hunk 3]
```diff
@@ -0,0 +1,39 @@
+--TEST--
+Test getCachedPath function
```
Comment (antecedent) [inline tests/get-cached-path.phpt:2]:
```suggestion
Test getCachedPath function (https://github.com/antecedent/patchwork/pull/170)
```

File: tests/get-cached-path.phpt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+$cachePath = str_replace('\\', '/', __DIR__ . '/cache');
+Patchwork\CodeManipulation\State::$cacheIndexFile = null;
+Patchwork\Config\State::$cachePath = $cachePath;
+
+if ( ! is_dir($cachePath)) {
+    mkdir($cachePath);
+}
+
+// Test getCachedPath function
```
Comment (antecedent) [inline tests/get-cached-path.phpt:19]:
```suggestion
// The expected path depends on the current implementation of getCachedPath().
```

**Review by antecedent — APPROVED** [review]

**Comment by jrfnl** [comment]
@antecedent Was just having another look at this, but the tests as it currently is, would not hit the PHP 8.4 deprecation notice (without the fix in place), so it's not safeguarding the change correctly. Digging in deeper now.

**Comment by jrfnl** [comment]
I've opened PR #171 to fix the test.

**Pull request merged**